avocado/core/sysinfo.py

```diff
@@ -202,21 +207,29 @@ def run(self, logdir):
         logf_path = os.path.join(logdir, self.logf)
         stdin = open(os.devnull, "r")
         stdout = open(logf_path, "w")
-        self.pipe = subprocess.Popen(shlex.split(self.cmd), stdin=stdin, stdout=stdout,
-                                     stderr=subprocess.STDOUT, shell=False, env=env)
+        try:
+            self.daemon_process = subprocess.Popen(shlex.split(self.cmd),
+                                                   stdin=stdin, stdout=stdout,
+                                                   stderr=subprocess.STDOUT,
+                                                   shell=False, env=env)
+        except OSError:
+            log.debug("Not logging  %s (command could not be run)" % self.cmd)
 
     def stop(self):
         """
         Stop daemon execution.
         """
-        retcode = self.pipe.poll()
-        if retcode is None:
-            process.kill_process_tree(self.pipe.pid)
-            retcode = self.pipe.wait()
-        else:
-            log.error("Daemon process '%s' (pid %d) terminated abnormally (code %d)",
-                      self.cmd, self.pipe.pid, retcode)
-        return retcode
+
+        if self.daemon_process is None:
```

Comment:
Here the check must be `if self.daemon_process is not None`. You need a daemon_process to stop. It's weird this didn't broke any test :thinking: .

Reply:
Yes, that's what I actually meant... :)

Reply:
@cacarrara ah sure i miss that :)